libraries/ms-helper/inc/thermistor.h

```diff
@@ -0,0 +1,22 @@
+#pragma once
+// Thermistor Reading interface for NXRT15XH103FA1B (10k Ohm NTC)
+// Requires ADC, GPIO and Interrupts to be initialized.
+#include "adc.h"
+#include "gpio.h"
+
+typedef struct {
+  uint32_t sibling_resistance_ohms;
+
+  ADCChannel adc_channel;
+} ThermistorStorage;
+
+// Initialize the thermistor interfaces
+StatusCode thermistor_init(ThermistorStorage *storage, GPIOAddress gpio_address,
```

Comment:
It would be nice if we could support both cases of resistor dividers (thermistor is top/bottom).

Also, please rename `gpio_address` to some more descriptive. What's it the gpio address for?

Reply:
Would't the calculation be the same regardless?

Reply:
Would it? I seem to remember going over this with you.